src/NServiceBus.Core/Serialization/SerializationFeature.cs

```diff
@@ -30,10 +30,16 @@ protected internal sealed override void Setup(FeatureConfigurationContext contex
 
             var defaultSerializer = CreateMessageSerializer(defaultSerializerAndDefinition, mapper, settings);
 
+            var additionalDeserializerDefinitions = context.Settings.GetAdditionalSerializers();
             var additionalDeserializers = new List<IMessageSerializer>();
-            foreach (var definitionAndSettings in context.Settings.GetAdditionalSerializers())
+
+            var additionalDeserializerDiagnostics = new List<Tuple<Type, IMessageSerializer>>();
+            foreach (var definitionAndSettings in additionalDeserializerDefinitions)
             {
-                additionalDeserializers.Add(CreateMessageSerializer(definitionAndSettings, mapper, settings));
+                var deserializer = CreateMessageSerializer(definitionAndSettings, mapper, settings);
+                additionalDeserializers.Add(deserializer);
+
+                additionalDeserializerDiagnostics.Add(new Tuple<Type, IMessageSerializer>(definitionAndSettings.Item1.GetType(), deserializer));
```

Comment:
why not already create the anonymous type here instead of creating a tuple just to map it later?

Reply:
Fixed